pkg/controller/testutils/testutils.go

```diff
@@ -32,3 +37,12 @@ func GetReconcileMetricCount(t *testing.T, controllerName, label string) float64
 	beforeCount := metricProto.GetCounter().GetValue()
 	return beforeCount
 }
+
+func StartTestingEnv() (*rest.Config, error) {
+	env = &envtest.Environment{}
+	return env.Start()
+}
+
+func CleanupTestingEnv() error {
```

Comment:
We are using the same env cross package. Some tests start the environment which I believe is idempotent but the stop condition isn't. When one package calls stop it will teardown the environment for all tests still running. Do our tests not run in parallel? I'm curious about why this works and why we don't need a separate Environment for each package.

Reply:
From my understanding, I don't think unit tests run in parallel, especially from different modules. In each module testenv Starts, runs unit tests withing module, Stops. Here is example when I run it locally with added `STARTING, STOPPING` logs:
<img width="761" alt="image" src="https://github.com/Azure/aks-app-routing-operator/assets/89222124/0cd9e162-0a39-49ee-8189-8cf5c63578a0">

We can run tests in parallel within single module by using -parallel.
Also, does go run unit tests concurrently ? Looking into that I'm not sure
https://github.com/golang/go/blob/master/src/testing/testing.go#L2025

Reply:
https://coderwall.com/p/rwgo_a/go-test-package-in-parallel

My understanding is that tests within a package are run sequentially by default but all packages are run in parallel.

Reply:
I found out that each `TestMain` function creates NEW in-memory testing environment with different `restConfig.Host` values.
I think that's why unit tests are not failing.